pkgpanda/util.py

```diff
@@ -413,3 +414,74 @@ def hash_checkout(item):
         return hash_list(list(item))
     else:
         raise NotImplementedError("{} of type {}".format(item, type(item)))
+
+
+def split_by_token(token_prefix, token_suffix, string_, strip_token_decoration=False):
```

Comment:
I yield to the current design.

My suggestion was not a new format. Basically keeping all the formats matched-up at a single place called format_types

```
format-types = {
'braces': ["{", "}"],
'double_square_brackets: ["[[", "]]"],
}
```

So, that when matching you only give the format type `split_by_token("braces", "{something}")` the module takes care of the rest.   This can prevent the possibility of a bad example that I gave in the previous comment.

Not a critical change request, so ok to drop it.